### PR TITLE
Refactor part of the koji_import plugin code

### PR DESCRIPTION
**Pull request by tkdchen**
This PR does not change the whole plugin implementation.

# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- n/a JSON/YAML configuration changes are updated in the relevant schema
- n/a Changes to metadata also update the documentation for the metadata
- n/a Pull request has a link to an osbs-docs PR for user documentation updates
- n/a New feature can be disabled from a configuration file

### CODE REVIEW

File: atomic_reactor/plugins/exit_koji_import.py [hunk 1]
```diff
@@ -145,6 +145,32 @@ def _worker_metadatas(self) -> Dict[str, Any]:
...
+        """
+        for worker_platform, metadata in self._worker_metadatas.items():
+            if platform is not None and worker_platform != platform:
+                continue
+            for output in metadata["output"]:
+                if _filter:
+                    match = True
+                    for key, value in _filter.items():
+                        match = match and output.get(key) == value
+                    if match:
```
Comment (chmeliik) [inline atomic_reactor/plugins/exit_koji_import.py:169]: opt. suggestion:
```python
if all(output.get(key) == value for key, value in _filter.items()):
```

File: atomic_reactor/plugins/exit_koji_import.py [hunk 2]
```diff
@@ -346,36 +371,37 @@ def set_group_manifest_info(self, extra):
...
-                            if 'digests' in annotations:
-                                digests = get_digests_map_from_annotations(annotations['digests'])
-                                instance['extra']['docker']['digests'] = digests
+            platform = "x86_64"
+            _, instance = next(
+                self._iter_work_metadata_outputs(platform, {"type": "docker-image"}),
+                (None, None),
+            )
+
+            if instance:
```
Comment (chmeliik) [inline atomic_reactor/plugins/exit_koji_import.py:375]: Nice! This removes 3 levels of indentation and makes it clearer what we're actually doing :tada: :taco:

**Review by chmeliik — COMMENTED** [review]
Nice! :rocket:

File: atomic_reactor/plugins/exit_koji_import.py (hunk 1 above)
Comment (tkdchen) [reply atomic_reactor/plugins/exit_koji_import.py]: Updated.

**Review by chmeliik — APPROVED** [review]
LGTM :+1:

**Review by lkolacek — APPROVED** [review]
Looks great!

**Pull request merged**